opal/mca/base/mca_base_var.c

```diff
@@ -1547,7 +1548,7 @@ int mca_base_var_register(const char *project_name, const char *framework_name,
     }
 
     OPAL_LIST_FOREACH_DECL(alias_item, &alias->component_aliases, mca_base_alias_item_t) {
-        mca_base_var_syn_flag_t flags_derived = flags;
```

Comment:
This changes the meaning of the code, making 0 the new default instead of `flags`.

Reply:
Yes.  See the commit message.  Setting the initial value to `flags` is actually a bug.

Reply:
The commit message is not clear enough on why this is a bug. What I understand from the code now is that all synonyms have minimalistic flags (basically just marked as synonyms) but without additional information about their capabilities. And I don't see the code that makes them inherit the flags of their legitimate synonym.

Reply:
The enum type is not a good unit to rely on. Do you have a case where this bug happens ?